image_client/picturae_importer.py

```diff
@@ -0,0 +1,986 @@
+
+        robjects.globalenv['r_dataframe_taxon'] = r_dataframe_tax
+
+        with open('taxon_check/R_TNRS.R', 'r') as file:
+            r_script = file.read()
+
+        robjects.r(r_script)
+
+        resolved_taxon = robjects.r['resolved_taxa']
+
+        resolved_taxon = robjects.conversion.rpy2py(resolved_taxon)
+
+        taxon_list = list(resolved_taxon['accepted_author'])
+
+        self.parent_author = taxon_list[0]
+
+
+    # think of finding way to make this function logic not so repetitive
+    def create_file_list(self):
+        """create_file_list: creates a list of imagepaths and barcodes for upload,
+                                after checking conditions established to prevent
+                                overwriting data functions
+        """
+        for index, row in self.record_full.iterrows():
+            if not row['image_valid']:
+                raise ValueError(f"image {row['image_path']} is not valid ")
+
+            elif not row['is_barcode_match']:
+                raise ValueError(f"image barcode {row['image_path']} does not match "
+                                 f"{row['CatalogNumber']}")
+
+            elif row['barcode_present'] and row['image_present']:
+                self.logger.warning(f"record {row['CatalogNumber']} and image {row['image_path']}"
+                                    f" already in database")
+
+            elif row['barcode_present'] and not row['image_present']:
+                self.logger.warning(f"record {row['CatalogNumber']} "
+                                    f"already in database, appending image")
+                self.image_list.append(row['image_path'])
+
+            elif not row['barcode_present'] and row['image_present']:
+                self.logger.warning(f"image {row['image_path']} "
+                                    f"already in database, appending record")
+                self.barcode_list.append(row['CatalogNumber'])
+            # create case so that if two duplicate rows , but with different images, upload both images,
+            # but not both rows
+            else:
+                self.image_list.append(row['image_path'])
+                self.barcode_list.append(row['CatalogNumber'])
+                self.barcode_list = list(set(self.barcode_list))
+                self.image_list = list(set(self.image_list))
+
+
+    def create_agent_list(self, row):
+        """create_agent_list:
+                creates a list of collectors that will be checked and added to agent/collector tables.
+                checks each collector first and last name against the database, and
+                then if absent, appends the new agent name to a list of dictionaries self.collector_list.
+           args:
+                row: a dataframe row containing collector name information
+        """
+        self.new_collector_list = []
+        self.full_collector_list = []
+
+        column_names = list(self.record_full.columns)
+        for i in range(1, 6):
```

Comment:
hardcoded value - could this be based on the len of the array? shoudl be len(thingy).